### PR TITLE
Support Django 1.11

### PR DESCRIPTION
**Pull request by Surgo**
It's time to support Django 1.11 🎉

### CODE REVIEW

File: django_sharding_library/fields.py [hunk 1]
```diff
@@ -129,7 +130,10 @@ def __init__(self, *args, **kwargs):
         if type(model_class) == str:
             app_label = model_class.split('.')[0]
             app = apps.get_app_config(app_label)
-            model_class = app.get_model(model_class[len(app_label) + 1:])
+            if (1, 11) > django.VERSION:
+                model_class = app.get_model(model_class[len(app_label) + 1:])
+            else:
+                model_class = app.get_model(model_class[len(app_label) + 1:], require_ready=False)
```
Comment (Surgo) [inline django_sharding_library/fields.py]: In Django 1.11 `django.core.exceptions.AppRegistryNotReady: Models aren't loaded yet.` is raised

https://docs.djangoproject.com/en/1.11/ref/applications/#django.apps.AppConfig.get_model

File: tests/models.py [hunk 2]
```diff
@@ -23,14 +23,18 @@ class ShardedModelIDs(TableStrategyModel):
 # the mixin provided in order to save the shard on the user.
 @shard_storage_config()
 class User(AbstractUser, ShardedByMixin):
-    pass
+    # Patch for Django 1.11
+    groups = None
```
Comment (Surgo) [inline tests/models.py]: In Django 1.11 `django.core.management.base.SystemCheckError: SystemCheckError: System check identified some issues:` is raised when run migrations.

```
ERRORS:
tests.PostgresShardUser.groups: (fields.E304) Reverse accessor for 'PostgresShardUser.groups' clashes with reverse accessor for 'User.groups'.
        HINT: Add or change a related_name argument to the definition for 'PostgresShardUser.groups' or 'User.groups'.
tests.PostgresShardUser.user_permissions: (fields.E304) Reverse accessor for 'PostgresShardUser.user_permissions' clashes with reverse accessor for 'User.user_permissions'.
        HINT: Add or change a related_name argument to the definition for 'PostgresShardUser.user_permissions' or 'User.user_permissions'.
tests.User.groups: (fields.E304) Reverse accessor for 'User.groups' clashes with reverse accessor for 'PostgresShardUser.groups'.
        HINT: Add or change a related_name argument to the definition for 'User.groups' or 'PostgresShardUser.groups'.
tests.User.user_permissions: (fields.E304) Reverse accessor for 'User.user_permissions' clashes with reverse accessor for 'PostgresShardUser.user_permissions'.
        HINT: Add or change a related_name argument to the definition for 'User.user_permissions' or 'PostgresShardUser.user_permissions'.
```

File: runtests.py [hunk 3]
```diff
@@ -65,7 +65,7 @@
     DATABASES=DATABASES,
     DATABASE_ROUTERS=['django_sharding_library.router.ShardedRouter'],
     AUTH_USER_MODEL='tests.User',
-    ROOT_URLCONF="django_sharding.urls",
+    ROOT_URLCONF="tests.urls",
```
Comment (Surgo) [inline runtests.py]: In Django 1.11 `ModuleNotFoundError: No module named 'django_sharding.urls'` is raised.

So add blank `./tests/urls.py`

**Comment by Surgo** [comment]
🙄

**Comment by JBKahn** [comment]
Totally missed this. Will take a look this week!

File: tests/models.py (hunk 2 above)
Comment (JBKahn) [inline tests/models.py]: I didn't have to do that in our application to make it run. Interesting...

File: tests/urls.py [hunk 4]
```diff
@@ -0,0 +1 @@
+urlpatterns = []
```
Comment (JBKahn) [inline tests/urls.py]: We may just be able to remove the reference from the settings file and not have a URLs file. Test that?

File: tests/models.py (hunk 2 above)
Comment (JBKahn) [inline tests/models.py]: Oh nevermind. I got it.

File: tests/urls.py (hunk 4 above)
Comment (Surgo) [inline tests/urls.py]: I removed `ROOT_URLCONF` and that worked fine :)

**Comment by Surgo** [comment]
🙄

**Comment by JBKahn** [comment]
Right.

**Comment by Surgo** [comment]
@JBKahn Thank you so much :)

**Pull request merged**